File: ArrayStack.cpp
#include <iostream>
using namespace std;

#define CAPACITY 10
  
class ArrayStack {
    int t;
  
public:
    int arr[CAPACITY]; 
  
    ArrayStack() { t = -1; }
    void push(int element);
    int pop();
    int top();
    bool isEmpty();
    int size();
    void print();
};


    void ArrayStack::push(int element)
    {
        if (t ==(CAPACITY - 1))
        {
            return;
        }
        else
        {
            t++;
            arr[t] = element;
        }
    }

  
int ArrayStack::pop()
{
        if (isEmpty())
        {
            return -1;
        }
        else
        {
            int temp = arr[t];
            arr[t] = -1;
            t--;
            return temp;
        }
    }
int ArrayStack::top()
{
        if (isEmpty())
        {
            return -1;
        }
        else
        {
            return arr[t];
        }
}
  
bool ArrayStack::isEmpty()
{
    return t == -1;
}

int ArrayStack::size(){
    return t + 1;
}

void ArrayStack::print(){
    for (int i = 0; i <= t; i++)
    {
        cout << arr[i] << "\t";
    }
    cout <<endl;
    
}

int main()
{
    ArrayStack stack;

    stack.push(2);
    stack.push(5);
    stack.push(10);

    stack.pop();
    
    stack.print();

    return 0;
}

File: ArrayStack.java
package javaapplication1;

/**
 *
 * @author Walaa Alfahmi
 */
public class ArrayStack<E>{
    private E[] array;
    private int top = -1;
    
    public ArrayStack(int capacity){
        array = (E[]) new Object[capacity];
    }
    
    public int size(){
        return top + 1;
    }
    public boolean isEmpty(){
        return top == -1;
    }
    
    public E top(){
        if (isEmpty()){
            return null;
        }
        else {
            return array[top];
        }
    }
    
    public E pop(){
        if (isEmpty()){
            return null;
        }
        else {
            E e = array[top];
            array[top] = null;
            top--;
            return e;
        }
    }
    
    public void push(E e){
        if(top == array.length - 1){
            return;
        }
        else {
            top++;
            array[top] = e;
        }
    }
    
    public void print(){
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + "\t");
        }
        System.out.println("");
        
    }
    
    public static void main(String[] args) {
        ArrayStack arStack = new ArrayStack(10);
        arStack.push("Walaa");
        arStack.push("Amal");
        arStack.push("Waad");
        arStack.print();
        arStack.push("Lama");
        arStack.print();
        System.out.println(arStack.size());
        System.out.println(arStack.pop());
        arStack.print();
        
    }
    
}
